src/fhirtypes/common.ts

```diff
@@ -854,6 +858,25 @@ export function replaceReferences<T extends AssignmentRule | CaretValueRule>(
       if (!value.reference.includes('/')) {
         assignedReference.reference = `${type}/${id}`;
       }
+      const typeFhir = fisher.fishForFHIR(type);
+      // if typeFhir is a logical, it needs to have can-be-target characteristic.
+      // this characteristic may appear as one of two extensions.
+      if (typeFhir?.kind === 'logical') {
```

Comment:
I'm not completely sure, but when we differentiate between profiles and logicals in FPL, it looks like we check that it is a `resourceType` of `'StructureDefinition'`, with a `kind` of `'logical'` and a derivation of `'specialization'`. I'm not sure if all that is needed here, but maybe?

Reply:
That's a good question. I was implementing this under the assumption that a profile of a logical model would also need to have the `can-be-target` characteristic, which it would typically inherit from the parent definition. If this assumption is correct, I can update the comment to state this more clearly.

Reply:
Ohhh that is a good point. I actually don't know the answer to that, but I guess I could test it out in the IG Publisher and see what happens.